### PR TITLE
Icon factory cleanups

### PR DESCRIPTION
**Pull request by jennej**
just me crusading against poor innocent files that haven't actually done anything wrong

### CODE REVIEW

File: core/ide/src/main/java/org/alice/stageide/gallerybrowser/shapes/BillboardDragModel.java [hunk 1]
```diff
@@ -42,23 +42,22 @@
...
 import org.alice.stageide.modelresource.ResourceNode;
 import org.lgna.croquet.SingleSelectTreeState;
 import org.lgna.croquet.Triggerable;
-import org.lgna.croquet.icon.IconFactory;
-import org.lgna.story.EmployeesOnly;
 import org.lgna.story.SBillboard;
 
 import java.util.UUID;
 
+// generic drag models.
```
Comment (dabeshouse) [inline core/ide/src/main/java/org/alice/stageide/gallerybrowser/shapes/BillboardDragModel.java]: Was including this comment an oversight?

File: core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java [hunk 2]
```diff
@@ -42,28 +42,38 @@
...
-  }
 
-  public static JointIconFactory getInstance() {
-    return SingletonHolder.instance;
-  }
+public class ShapeIconFactory extends ResolutionIndependentIconFactory {
+  Constructor<? extends ShapeIcon> iconConstructor;
 
-  private JointIconFactory() {
+  public ShapeIconFactory(Class<? extends ShapeIcon> cls)  {
```
Comment (dabeshouse) [inline core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java:58]: I was starting to suggest replacing the Class with a direct reference to the new method, but once I started down that path I saw how much more of the icon factory hierarchy can be collapsed or eliminated.
I think I will offer it as possible next step PR.

To be overly precise, the type would become `Function<Dimension, ShapeIcon>`, called with things like `DiscIcon::new` and invoked with `iconConstructor.apply(size)`.

File: core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java [hunk 3]
```diff
@@ -42,28 +42,38 @@
...
-public class JointIconFactory extends ResolutionIndependantIconFactory {
-  private static class SingletonHolder {
-    private static JointIconFactory instance = new JointIconFactory();
-  }
 
-  public static JointIconFactory getInstance() {
-    return SingletonHolder.instance;
-  }
+public class ShapeIconFactory extends ResolutionIndependentIconFactory {
+  Constructor<? extends ShapeIcon> iconConstructor;
```
Comment (dabeshouse) [inline core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java:56]: Could be final ;)

**Review by dabeshouse — APPROVED** [review]

File: core/ide/src/main/java/org/alice/stageide/gallerybrowser/shapes/BillboardDragModel.java (hunk 1 above)
Comment (jennej) [reply core/ide/src/main/java/org/alice/stageide/gallerybrowser/shapes/BillboardDragModel.java]: oops, yeah 😅

File: core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java (hunk 2 above)
Comment (jennej) [reply core/ide/src/main/java/org/alice/stageide/icons/ShapeIconFactory.java:58]: I support all ideas to dismantle the icon factory industrial complex!

**Pull request merged**